website/docs/basics.md

```diff
@@ -0,0 +1,214 @@
+
+##### Stack Outputs
+
+Stacks can export values as [Stack Outputs](https://www.pulumi.com/docs/concepts/stack/#outputs).
+These outputs are shown by Pulumi CLI commands, and are displayed in the Pulumi Cloud, and can be accessed 
+programmatically using [Stack References](#stack-references).
+
+To export values from a stack in Besom, use the [`Pulumi.exports`](exports) function in your program.
+
+##### Stack References
+
+[Stack References](https://www.pulumi.com/docs/concepts/stack/#stackreferences) allow you to use outputs from other 
+[stacks](#stacks) in your [program](#programs).
+
+To reference values from another stack, create an instance of the `StackReference` type using the fully qualified 
+name of the stack as an input, and then read exported stack outputs by their name.
+
+`StackReference` is not implemented yet in Besom, coming soon.
+
+#### Resources
+
+Resources are the primary [construct of Pulumi](https://www.pulumi.com/docs/concepts/resources/) programs.
+Resources represent the fundamental units that make up your infrastructure, such as a compute instance,
+a storage bucket, or a Kubernetes cluster.
+
+Resources are defined using a [**resource constructor**](constructors). Each resource in Pulumi has:
+- a [logical name and a physical name](https://www.pulumi.com/docs/concepts/resources/names/#resource-names)
+  The logical name establishes a notion of identity within Pulumi, and the physical name is used as identity by the provider
+- a [resource type](https://www.pulumi.com/docs/concepts/resources/names/#types), which identifies the provider and the kind of resource being created
+- [URN](https://www.pulumi.com/docs/concepts/resources/names/#types), which is an automatically constructed globally unique identifier for the resource.
+
+Each resource also can have:
+- a set of [arguments](https://www.pulumi.com/docs/concepts/resources/properties/) that define the behavior of the resulting infrastructure
+- and a set of [options](https://www.pulumi.com/docs/concepts/options/) that control how the resource is created and managed by the Pulumi engine.
+
+Every is managed by a [provider](https://www.pulumi.com/docs/concepts/resources/providers/) which is a plugin that
+provides the implementation details. If not specified explicitly, the default provider is used.
+Providers can be configured using [provider configuration](https://www.pulumi.com/docs/concepts/resources/providers/#explicit-provider-configuration).
+
+#### Inputs and Outputs
+
+Inputs and Outputs are the primary [asynchronous data types in Pulumi](https://www.pulumi.com/docs/concepts/inputs-outputs/), 
+and they signify values that will be provided by the engine later, when the resource is created and its properties can be fetched.
+`Input[A]` type is an alias for `Output[A]` type used by [resource](#resources) arguments.
+
+Outputs are values of type `Output[A]` and behave very much like [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)). 
```

Comment:
"like" monads or Besom Output is an actual monad?

Reply:
Hmmm, `Output` has a bit of a complicated semantic. I think that it wouldn't be a crime to call them monads. The only benefit we get from calling it a monad is the analogy. I don't see anyone using generic functions over Monads for `Output`.
Maybe a useful analogy would be to say that it is besom's effect type?

Reply:
OK then, I'll leave as-is, this is the language Pulumi is using in their docs "like monad"